diff --git a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/system-context.ts b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/system-context.ts
index 5750147..fc95d4a 100644
--- a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/system-context.ts
+++ b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/system-context.ts
@@ -17,6 +17,17 @@ type ScrapeResult = {
   result: string;
 };
 
+const textPartsToContent = (parts: UIMessage["parts"]) => {
+  return parts
+    .map((part) => {
+      if (part.type === "text") {
+        return part.text;
+      }
+      return "";
+    })
+    .join("");
+};
+
 const toQueryResult = (query: QueryResultSearchResult) =>
   [`### ${query.date} - ${query.title}`, query.url, query.snippet].join("\n\n");
 
@@ -49,7 +60,7 @@ export class SystemContext {
     return this.messages
       .map((message) => {
         const role = message.role === "user" ? "User" : "Assistant";
-        return `<${role}>${message.content}</${role}>`;
+        return `<${role}>${textPartsToContent(message.parts)}</${role}>`;
       })
       .join("\n\n");
   }
