diff --git a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/api/chat/route.ts b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/api/chat/route.ts
index 1b90ffe..094a9bb 100644
--- a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/api/chat/route.ts
+++ b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/api/chat/route.ts
@@ -1,5 +1,8 @@
-import type { UIMessage } from "ai";
-import { createDataStreamResponse, appendResponseMessages } from "ai";
+import {
+  createUIMessageStream,
+  createUIMessageStreamResponse,
+  type UIMessage,
+} from "ai";
 import { auth } from "~/server/auth";
 import { upsertChat } from "~/server/db/queries";
 import { eq } from "drizzle-orm";
@@ -64,11 +67,12 @@ export async function POST(request: Request) {
     userId: session.user.id,
   });
 
-  return createDataStreamResponse({
+  const stream = createUIMessageStream({
     execute: async (dataStream) => {
       // If this is a new chat, send the chat ID to the frontend
       if (!chatId) {
-        dataStream.write({
+        dataStream.writer.write({
+          // TODO: fix this type error
           type: "data",
 
           value: [
@@ -86,6 +90,7 @@ export async function POST(request: Request) {
         messages,
         onFinish: async ({ response }) => {
           // Merge the existing messages with the response messages
+          // TODO: fix this type error
           const updatedMessages = appendResponseMessages({
             messages,
             responseMessages: response.messages,
@@ -121,11 +126,13 @@ export async function POST(request: Request) {
         },
       });
 
-      result.mergeIntoDataStream(dataStream);
+      dataStream.writer.merge(result.toUIMessageStream());
     },
     onError: (e) => {
       console.error(e);
       return "Oops, an error occurred!";
     },
   });
+
+  return createUIMessageStreamResponse({ stream });
 }
