diff --git a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/deep-search.ts b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/deep-search.ts
index 3c326f5..875cd23 100644
--- a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/deep-search.ts
+++ b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/deep-search.ts
@@ -26,5 +26,5 @@ export async function askDeepSearch(messages: UIMessage[]) {
   // the stream will never finish
   await result.consumeStream();
 
-  return await result.text.text;
+  return await result.text;
 }
