diff --git a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/server/db/queries.ts b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/server/db/queries.ts
index 6064b36..83d6c43 100644
--- a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/server/db/queries.ts
+++ b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/server/db/queries.ts
@@ -68,7 +68,6 @@ export const getChat = async (opts: { userId: string; chatId: string }) => {
       id: message.id,
       role: message.role,
       content: message.parts,
-      annotations: message.annotations ?? [],
     })),
   };
 };
