diff --git a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx
index 0f843bd..b4e0c0d 100644
--- a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx
+++ b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx
@@ -29,24 +29,15 @@ export const ChatPage = ({
       chatId,
     },
     initialMessages,
+    onData: (data) => {
+      if (data.type === "data-new-chat-created") {
+        router.push(`?id=${data.data.chatId}`);
+      }
+    },
   });
 
   const isLoading = status === "streaming";
 
-  useEffect(() => {
-    const lastMessage = messages[messages.length - 1];
-    const mostRecentPartOfMessage =
-      lastMessage?.parts?.[lastMessage.parts.length - 1];
-    if (
-      mostRecentPartOfMessage &&
-      mostRecentPartOfMessage.type === "data-new-chat-created"
-    ) {
-      router.push(`?id=${mostRecentPartOfMessage.data.chatId}`);
-    }
-  }, [messages, router]);
-
-  const [input, setInput] = useState("");
-
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
 
