diff --git a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx
index 3ad14f5..036d698 100644
--- a/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx
+++ b/courses/01-deepsearch-in-typescript/00-apps/05-day-8-app/src/app/chat.tsx
@@ -39,24 +39,15 @@ export const ChatPage = ({
 
   const { messages, sendMessage, status } = useChat<OurMessage>({
     chat,
+    onData: (data) => {
+      if (data.type === "data-new-chat-created") {
+        router.push(`?id=${data.data.chatId}`);
+      }
+    },
   });
 
   const isLoading = status === "streaming";
 
-  useEffect(() => {
-    const lastMessage = messages[messages.length - 1];
-    const mostRecentPartOfMessage =
-      lastMessage?.parts?.[lastMessage.parts.length - 1];
-
-    console.log(mostRecentPartOfMessage);
-    if (
-      mostRecentPartOfMessage &&
-      mostRecentPartOfMessage.type === "data-new-chat-created"
-    ) {
-      router.push(`?id=${mostRecentPartOfMessage.data.chatId}`);
-    }
-  }, [messages, router]);
-
   const [input, setInput] = useState("");
 
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
